packages/plugin-ext/src/hosted/node/plugin-host-rpc.ts

```diff
@@ -128,40 +129,46 @@ export class PluginHostRPC {
                     console.error(e);
                 }
             },
-            init(raw: PluginMetadata[]): [Plugin[], Plugin[]] {
+            async init(raw: PluginMetadata[]): Promise<[Plugin[], Plugin[]]> {
```

Comment:
are we forced to move from sync to async there ?
From what I see it's to read package.json files but can't it be delayed later as well and only used for frontends not backend yet.

Reply:
Only if we replace with fs sync read since `rawModel` is always used synchronously. But using sync fs operations is discouraged for perf reasons.

Reply:
I was thinking on more lazy loading with later asynchronous operations, not during init phase, not using sync read.

Reply:
I will try to move it just before calling plugin initialization script.

Reply:
I cannot move it because we call `self.initContext(backendInitPath, plugin);` in this method and raw model should be already there for plugin-vscode-init: https://github.com/theia-ide/theia/blob/dd1d2a0c9c4f9e0326771ab2b4027b8ba28c4181/packages/plugin-ext-vscode/src/node/plugin-vscode-init.ts#L47-L49